merge_checks.py

```diff
@@ -2,16 +2,16 @@
 import subprocess
 
 
-def main(base_ref: str, head_ref: str):
-    print(f"Checking out BASE_REF ({base_ref}) and HEAD_REF ({head_ref})...")
-
+def main(head_hash: str, base_ref: str):
+    print(f"Checking out {base_ref}...")
     subprocess.run(f"git fetch --depth=1 origin {base_ref}", check=True, shell=True)
-    subprocess.run(f"git fetch origin {head_ref}", check=True, shell=True)
-
     base_revision = subprocess.run(f"git rev-parse origin/{base_ref}", check=True, shell=True, capture_output=True)
-    head_revision = subprocess.run(f"git rev-parse origin/{head_ref}", check=True, shell=True, capture_output=True)
 
-    base_hash, head_hash = (output.decode().strip() for output in (base_revision.stdout, head_revision.stdout))
+    base_hash = base_revision.stdout.decode().strip()
+
+    if head_hash == base_hash:
+        print(f"HEAD identical with {base_ref}, no commits to check")
+        exit(0)
```

Comment:
Ich habe das leider vermutlich selbst im Stress verursacht, aber eigentlich `exit()` soll in Programmen gar nicht benutzt werden - das ist ein Helper aus `site`-Modul für interactive Skripten - so ein bisschen wie Zeugs aus Prelude in Haskell.

Stattdessen soll `exit` von `sys` benutzt werden - `import sys; sys.exit(0)`. Der praktische Nachteil von `exit` aus `site` liegt daran, dass Python-Programme werden mit `-S` gar nicht laufen:

```
$ python3 -S -c "exit()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
NameError: name 'exit' is not defined
```

Reply:
Danke für die Erklärung, @zyv. Ich habe das in https://github.com/moneymeets/action-merge-checks/pull/4 angepasst.